### DISCUSSION TITLE
Cross account parameters

### DISCUSSION THREAD

**Discussion by andremantaswow** [aws/aws-dotnet-extensions-configuration#194] [Q&A] [answered]
We're loading configs from Parameter Store and would like to use this package. The issue is that they are on another account (account B) from where our lambda is running (account A). 

This is what we are doing so far:
- On account B we've created a role that has access to the parameters and granted access to be assumed by lambdas in account A.
- On the lambda in account A we assume the role and use those credentials to initialize a `AmazonSimpleSystemsManagementClient` and call `GetParametersByPathAsync`.

Is it possible to achieve the same with this package?

Thanks.

**Comment by ashishdhingra** [comment, marked as answer]
@andremantaswow Good afternoon. Thanks for opening the discussion. Per [Working with shared parameters in Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/parameter-store-shared-parameters.html), to share a parameter, it needs to be added to resource share. Thereafter, it could be accessed via ARN, e.g using AWS CLI:
```
aws ssm get-parameter \
    --name arn:aws:ssm:us-east-2:123456789012:parameter/MySharedParameter
```

This package uses [GetParametersByPath]9https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetParametersByPath.html) service API call which could only get parameters that start at a certain path. As a result, accessing shared parameters might not be currently supported by this package.

Thanks,
Ashish
